torch/onnx/utils.py

```diff
@@ -1245,42 +1240,68 @@ def export_to_pretty_string(
+    model,
+    args,
+    training: _C_onnx.TrainingMode = _C_onnx.TrainingMode.EVAL,
+    opset_version: Optional[int] = None,
+) -> Tuple[_C.Graph, List[str]]:
+    """Returns an approximated list of all ops that are yet supported by :mod:`torch.onnx`.
+
+    The list is approximated because some ops may be removed during the conversion
+    process and don't need to be converted. Some other ops may have partial support
+    that will fail conversion with particular inputs. Please open a Github Issue
+    for op support requests.
 
     Args:
-        model: Same as corresponding arg to torch.onnx.export.
-        args: Same as corresponding arg to torch.onnx.export.
-        training: Same as corresponding arg to torch.onnx.export.
-        opset_version: Same as corresponding arg to torch.onnx.export.
+        model: Same as the `model` parameter in :func:`torch.onnx.export`.
+        args: Same as the `args` parameter in :func:`torch.onnx.export`.
+        training: Same as the `training` parameter in :func:`torch.onnx.export`.
+        opset_version: Same as the `opset_version` parameter in :func:`torch.onnx.export`.
 
     Returns:
-        Tuple[torch._C.Graph, List[str]], where the list includes the names
-        of the unconvertible ops.
+        The JIT graph and a list of unconvertible ops in the format of "domain::op".
     """
 
     opset_version = opset_version or _constants.ONNX_DEFAULT_OPSET
     symbolic_helper._set_opset_version(opset_version)
-    # operator_export_type is set to ONNX_FALLTHROUGH by default so that if an op is not supported
-    # in ONNX, fall through will occur and export the operator as is, as a custom ONNX op.
-    with exporter_context(model, training, False):
-        args = _decide_input_format(model, args)
-        graph, params_dict, torch_out = _model_to_graph(
-            model,
-            args,
-            # So that if an op connot be converted to ONNX, it will be kept
-            # as-is rather than cause a failure.
-            operator_export_type=_C_onnx.OperatorExportTypes.ONNX_FALLTHROUGH,
-        )
-    unsupported_ops = list()
-    supported_namespaces = {"onnx", "prim", "quantized"}
+
+    try:
+        with exporter_context(model, training, verbose=False):
+            # Create a mostly clean JIT graph that contains the plain aten and
+            # other ops we can check with the symbolic registry.
+            # NOTE: We don't want to actually convert any ops to ONNX or run any
+            # symbolic functions because there is a higher chance that a pass
+            # fails or an unconvertible op messes up the graph during ONNX conversion.
+            # This way we can always generate a list just by looking at the names
+            # of the ops in the graph.
+            args = _decide_input_format(model, args)
+            model = _pre_trace_quant_model(model, args)
+            graph, _, _, module = _create_jit_graph(model, args)
```

Comment:
How do you make sure that slipping passes won't make error in the graph? Whether it passes CI?